### PR TITLE
Add Logging Functionality, @amplitude/utils, and LogLevel to options when initializing amplitude client

### PR DESCRIPTION
**Pull request by kelvin-lu**

### CODE REVIEW

File: packages/utils/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<p align="center">
+  <a href="https://amplitude.com" target="_blank" align="center">
+    <img src="https://static.amplitude.com/lightning/46c85bfd91905de8047f1ee65c7c93d6fa9ee6ea/static/media/amplitude-logo-with-text.4fb9e463.svg" width="280">
+  </a>
+  <br />
+</p>
+
+[![npm version](https://badge.fury.io/js/%40amplitude%2Ftypes.svg)](https://badge.fury.io/js/%40amplitude%2Ftypes)
```
Comment (haoliu-amp) [inline packages/utils/README.md:8]: Make sure to replace this after this package getting published.

File: packages/utils/package.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    "node": ">=6"
+  },
+  "main": "dist/index.js",
+  "module": "esm/index.js",
+  "types": "dist/index.d.ts",
+  "publishConfig": {
+    "access": "public"
+  },
+  "dependencies": {
+    "@amplitude/types": "^0.3.2"
```
Comment (haoliu-amp) [inline packages/utils/package.json:19]: Does `utils` depend on `types`?

File: packages/utils/src/logger.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  /** JSDoc */
+  public error(...args: any[]): void {
+    if (!this._logLevel <= LogLevel.ERROR) {
+      return;
+    }
+    global.console.error(`${PREFIX}[Error]: ${args.join(' ')}`); // tslint:disable-line:no-console
+  }
+}
+
+// Ensure we only have a single logger instance, even if multiple versions of @sentry/utils are being used
```
Comment (haoliu-amp) [inline packages/utils/src/logger.ts:55]: 👍

File: packages/utils/src/logger.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { LogLevel } from '@amplitude/types';
+import { getGlobalObject } from './misc';
+
+// TODO: Type the global constant
+const global = getGlobalObject();
+
+/** Prefix for logging strings */
+const PREFIX = 'Amplitude Logger ';
+
+/** JSDoc */
```
Comment (ManThursday) [inline packages/utils/src/logger.ts:10]: What's up with all the JSDoc stuff?

File: packages/utils/src/logger.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  /** JSDoc */
+  public error(...args: any[]): void {
+    if (this._logLevel < LogLevel.Error) {
+      return;
+    }
+    global.console.error(`${PREFIX}[Error]: ${args.join(' ')}`); // tslint:disable-line:no-console
+  }
+}
+
+// Ensure we only have a single logger instance, even if multiple versions of @amplitude/utils are being used
```
Comment (ManThursday) [inline packages/utils/src/logger.ts:55]: Why is this important?

**Pull request merged**